### ISSUE TITLE
Add support for GL702VSK

### ISSUE THREAD

**Issue by bonezinger** [dominiksalvet/asus-fan-control#89]
TLDR: It works, but fan speed is capped at 6k.

**Environment**
Linux Mint 20.1
Asus-fan-control 3.10

**Basic configuration**
If only temperatures were set and they worked, add output of:
* `sudo asus-fan-control get-temps` - 45 46 50 51 52 53 58 62
* * NOTE - I set a low threshold because the GL702 has a serious overheating problem.
* `sudo asus-fan-control model-info`
```
name GL702VSK
addresses 1335
default-temps 55 60 62 65 68 72 76 80
tested false
```

**Additional context**
The fans reach a maximum of 6000 RPM with the utility. On `xsensors`, the "progress bar" of the fan speed sits around the middle while at 5k. On Windows, the Asus ROG program can boost the fan over that 6k limit, but doesn't show the actual spin speed. Judging from the xsensors, it must be around 10k RPM. So there is something "locking" the maximum fan speed.

Of note:
* The fans are from the graphics card (nvidia)
* `pwmconfig` and `lm-sensors` cannot find the fan

I'm willing to try any alterations and tests to see if we can get rid of the fan speed limit.

**Comment by dominiksalvet** [comment]
Ok. So first question. Does your laptop have multiple fans? Are all driven equally?

My suggestion about the max cap would be to have a look at the utility I have created [afc-acout](https://github.com/dominiksalvet/afc-scout). You can see your default address ranges without using any arguments. Then, you can see whether there is something above that using start and end address arguments... Maybe your laptop uses more temperatures to drive your fan by (AFC has been prepared for that! 😁)... Feel free to ask for help if you need. 👍

**Comment by bonezinger** [comment]
It has 2 fans and they're driven equally.
I'll try that afc-acout tomorrow and say if it works.

**Comment by dominiksalvet** [comment]
> I don't know what the latter is supposed to represent.

For currently supported dual-fan laptops, they represent the temps by which the second fan is driven.

> It has 2 fans and they're driven equally (or so I think).

We can easily find out... try high temperatures temporarily and all fans of the laptop should turn off. E.g., `sudo asus-fan-control set-temps 70 71 72 73 74 75 76 77`. Then, very low temperatures and all should go at their max. E.g., `sudo asus-fan-control set-temps 30 31 32 33 34 35 36 37`. For more info, see [our wiki](https://github.com/dominiksalvet/asus-fan-control/wiki).

Let me know how it went... :smile:

**Comment by bonezinger** [comment]
```
:~$ sudo asus-fan-control set-temps 70 71 72 73 74 75 76 77
:~$ sudo afc-scout
------ base address 1335 ------
[1334] = 12
[1335] = 70 <- possible start
[1336] = 71
[1337] = 72
[1338] = 73
[1339] = 74
[1340] = 75
[1341] = 76
[1342] = 77 <- possible end
[1343] = 12
------ base address 1431 ------
[1430] = 12
[1431] = 45 <- possible start
[1432] = 50
[1433] = 56
[1434] = 62
[1435] = 68
[1436] = 73
[1437] = 76
[1438] = 80 <- possible end
[1439] = 12
```

Fan speed at ~52° (idling) is 2700. The fans speed up to 6k if it reaches the set temps.

```
:~$ sudo asus-fan-control set-temps 30 31 32 33 34 35 36 37
:~$ sudo afc-scout
------ base address 1335 ------
[1334] = 12
[1335] = 30 <- possible start
[1336] = 31
[1337] = 32
[1338] = 33
[1339] = 34
[1340] = 35
[1341] = 36
[1342] = 37 <- possible end
[1343] = 12
------ base address 1431 ------
[1430] = 12
[1431] = 45 <- possible start
[1432] = 50
[1433] = 56
[1434] = 62
[1435] = 68
[1436] = 73
[1437] = 76
[1438] = 80 <- possible end
[1439] = 12
```

Fan speed at ~49°C (idling) was 6000 for a while. It eventually slowed to around 5k. Now, the weirdest thing, once I put it under stress, with xsensors noting the temps over 80°C (i could feel it was hot as hell with my hand near the screen), it took a while to speed the fans up again.

Is there a way to do a similar settemp for that second base address?

Even with the low temperature thresholds, it never spun faster than 6k. Even on Windows, it only ever reaches its real max speed if you manually do so in the ROG Center.

**Comment by dominiksalvet** [comment]
> Even with the low temperature thresholds, it never spun faster than 6k.

Please, let's focus on a major issue first -- whether both fans are driven equally. We can examine the 6k/10k RPM issue later.

> Is there a way to do a similar settemp for that second base address?

You can control both fans at once using environment variables. Use `sudo FALLBACK_ADDRS='1335 1431' FALLBACK_TEMPS='<temps>' asus-fan-control set-temps default`. Please note that there are very few checks before calling ACPI compared to the previous method. Be always sure you are passing the correct number of temperatures.

However, once you test it this way and it works, we will add the support for laptop and once AFC is updated, it will do this internally using the previous, more convenient, method.

For more information, see [contributing.md](https://github.com/dominiksalvet/asus-fan-control/blob/master/contributing.md).

**Comment by bonezinger** [comment]
```
~$ sudo FALLBACK_ADDRS='1335 1431' FALLBACK_TEMPS='40 43 46 49 52 55 58 61' asus-fan-control set-temps default
~$ sudo afc-scout 
------ base address 1335 ------
[1334] = 12
[1335] = 40 <- possible start
[1336] = 43
[1337] = 46
[1338] = 49
[1339] = 52
[1340] = 55
[1341] = 58
[1342] = 61 <- possible end
[1343] = 12
------ base address 1431 ------
[1430] = 12
[1431] = 40 <- possible start
[1432] = 43
[1433] = 46
[1434] = 49
[1435] = 52
[1436] = 55
[1437] = 58
[1438] = 61 <- possible end
[1439] = 12

```

Everything seems to be working fine, the fans speed up and slow down properly as the temperature varies. It doesn't look like the second address controls any fans, i've set the first's temperatures to something much higher (70 and up at the lowest), xsensors showed the fans on a slow rotation, ~3k rpm, while around 65°C.

**Comment by dominiksalvet** [comment]
Alright, so maybe the first address range controls both fans. So we will add only that range to AFC. I will do it in the following days (I am quite busy now, sorry).

Please, do not rely much on RPM sensors on Linux, they may be correct as well as they may not. E.g., for my UX430UA, they do not work at all. Quiet environment and your ears is the best tool for that... 😉

---

The investigation of 6K RPM limit may continue in the original discussion to keep this issue clean.

**Comment by KubeKing** [comment]
@bonezinger Did you ever find a solution to "boosting" the fan speed. The AFC controller does work on my unit (GL702VSK), but it's not able to reach the same speeds as the Windows ROG controller did when set to 100%.
